cmd/vocab-service/neo4j.go

```diff
@@ -0,0 +1,331 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"strings"
+
+	neo "github.com/johnnadratowski/golang-neo4j-bolt-driver"
+)
+
+var DefaultMaxConn = 20
+
+type service struct {
+	url  string
+	pool neo.DriverPool
+}
+
+func (s *service) Match(cxt context.Context, vocab, pattern string, res chan<- *Class) error {
+	if vocab == "" {
+		return errors.New("vocab cannot be empty")
+	}
+
+	if pattern == "" {
+		return errors.New("pattern cannot be empty")
+	}
+
+	// Case insensitive.
+	pattern = strings.ToLower(pattern)
+
+	conn, err := s.pool.OpenPool()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	stmt, err := conn.PrepareNeo(`
+		MATCH (v:Vocabulary {id: {vocab}})<-[:classOf]-(c:Class)
+		WHERE lower(c.label) =~ {pattern}
+			OR any(syn in c.synonyms where lower(syn) =~ {pattern})
+		RETURN c.id, v.id, c.label, c.code, c.synonyms
+	`)
+	if err != nil {
+		return err
+	}
+
+	rows, err := stmt.QueryNeo(map[string]interface{}{
```

Comment:
I guess this can't use the `traversal` function because it has different parameters, right?